WordPress/Classes/Models/ReaderPost.m

```diff
@@ -178,5 +179,66 @@ - (NSURL *)featuredImageURLForDisplay
     return [self featuredImageURL];
 }
 
+- (SourceAttributionStyle)sourceAttributionStyle
+{
+    if (!self.sourceAttribution) {
+        return SourceAttributionStyleNone;
+    }
+    if ([self.sourceAttribution.attributionType isEqualToString:SourcePostAttributionTypePost]) {
+        return SourceAttributionStylePost;
+    } else if ([self.sourceAttribution.attributionType isEqualToString:SourcePostAttributionTypeSite]) {
+        return SourceAttributionStyleSite;
+    } else {
+        return SourceAttributionStyleNone;
+    }
+}
+
+- (NSString *)sourceAuthorNameForDisplay
+{
+    if (!self.sourceAttribution) {
```

Comment:
I believe this check is not needed, actually. If the sourceAttribution is nil, this will return nil anyways